### PR TITLE
Add documentation for GKE workload identity

### PR DESCRIPTION
**Pull request by ddgenome**
Add documentation and link to blog post for using GKE workload
identity to authenticate against Google Cloud DNS.

Per request in jetstack/cert-manager#3009

Signed-off-by: David Dooling <dooling@gmail.com>

### CODE REVIEW

File: content/en/docs/configuration/acme/dns01/google.md [hunk 1]
```diff
@@ -25,10 +25,12 @@ DNS01 challenge. To enable this, a GCP service account must be created with the
 > Console.
 
 ```bash
-$ export PROJECT_ID=myproject-id
+$ PROJECT_ID=myproject-id
```
Comment (meyskens) [inline content/en/docs/configuration/acme/dns01/google.md:28]: Why is export dropped here?

File: content/en/docs/configuration/acme/dns01/google.md [hunk 2]
```diff
@@ -122,3 +130,104 @@ spec:
...
+The cert-manager component that needs to modify DNS records is the pod created
+as part of the cert-manager deployment. The [standard methods for deploying
+cert-manger to Kubernetes](../../../installation/kubernetes.md) create the
+cert-manager deployment in the cert-manager namespace and its pod spec specifies
+it runs under the cert-mananger service account. To link the GSA you created
+above to the cert-manager KSA in the cert-manager namespace in your GKE cluster,
+run the following command.
+
+```bash
+$ gcloud iam service-accounts add-iam-policy-binding \
```
Comment (meyskens) [inline content/en/docs/configuration/acme/dns01/google.md:162]: Have you tested if this stays when upgrading cert-manager? I see you apply an annotation below which could be deleted on upgrade

File: content/en/docs/configuration/acme/dns01/google.md [hunk 3]
```diff
@@ -122,3 +130,104 @@ spec:
...
+`NAMESPACE` is the namespace of the service account and `SERVICE_ACCOUNT` is the
+name of the service account.
+
+### Link KSA to GSA in Kubernetes
+
+After deploying cert-manager, add the proper workload identity annotation to the
+cert-manager service account.
+
+```bash
+$ kubectl annotate serviceaccount --namespace=cert-manager cert-manager \
```
Comment (meyskens) [inline content/en/docs/configuration/acme/dns01/google.md:179]: Is there a way to do this with the Helm chart also?

File: content/en/docs/configuration/acme/dns01/google.md [hunk 4]
```diff
@@ -122,3 +130,104 @@ spec:
...
+### Create an Issuer That Uses CloudDNS
+
+Next, create an `Issuer` (or `ClusterIssuer`) with a `clouddns` provider. An
+example `Issuer` manifest can be seen below with annotations. Note that the
+issuer does not include a `serviceAccountSecretRef` property. Excluding this
+instructs cert-manager to use the default credentials provided by GKE workload
+identity.
+
+```yaml
+apiVersion: cert-manager.io/v1alpha2
```
Comment (meyskens) [inline content/en/docs/configuration/acme/dns01/google.md:197]:
```suggestion
apiVersion: cert-manager.io/v1
```

We're releasing v1 tomorrow with the v1 api so we're already keeping the docs up to date before merging

File: content/en/docs/configuration/acme/dns01/google.md [hunk 5]
```diff
@@ -122,3 +130,104 @@ spec:
...
+kind: Issuer
+metadata:
+  name: example-issuer
+spec:
+  acme:
+    ...
+    solvers:
+    - dns01:
+        clouddns:
```
Comment (meyskens) [inline content/en/docs/configuration/acme/dns01/google.md:206]:
```suggestion
        cloudDNS:
```

changed in v1, well k8s YAML is not case sensitive but just to be correct

File: content/en/docs/configuration/acme/dns01/google.md [hunk 6]
```diff
@@ -122,3 +130,104 @@ spec:
...
+```
+
+For more information about `Issuers` and `ClusterIssuers`, see
+[Configuration](../../../).
+
+Once an `Issuer` (or `ClusterIssuer`) has been created successfully, a
+`Certificate` can then be added to verify that everything works.
+
+```yaml
+apiVersion: cert-manager.io/v1alpha2
```
Comment (meyskens) [inline content/en/docs/configuration/acme/dns01/google.md:218]:
```suggestion
apiVersion: cert-manager.io/v1
```

**Comment by meyskens** [comment]
Great job! Looks very good, just a few small things before merging

File: content/en/docs/configuration/acme/dns01/google.md (hunk 1 above)
Comment (ddgenome) [reply content/en/docs/configuration/acme/dns01/google.md:28]: Because it serves no purpose. Nowhere is `PROJECT_ID` used at an environment variable. It is just used as a shell variable.

File: content/en/docs/configuration/acme/dns01/google.md (hunk 2 above)
Comment (ddgenome) [reply content/en/docs/configuration/acme/dns01/google.md:162]: The `gcloud` commands persist when upgrading cert-manager.

Comment (ddgenome) [reply content/en/docs/configuration/acme/dns01/google.md:162]: If you use `kubectl apply` to upgrade, then the annotation added later should remain as well. If you delete and recreate, then you would need to add the annotation again.

File: content/en/docs/configuration/acme/dns01/google.md (hunk 3 above)
Comment (ddgenome) [reply content/en/docs/configuration/acme/dns01/google.md:179]: The helm chart offers the `serviceAccount.annotations` parameter for this purpose.

Comment (meyskens) [reply content/en/docs/configuration/acme/dns01/google.md:179]: Can we document that also for those who use helm?

File: content/en/docs/configuration/acme/dns01/google.md (hunk 2 above)
Comment (meyskens) [reply content/en/docs/configuration/acme/dns01/google.md:162]: okay, we recommend to use applies so that would work, also with the helm tag it solves that too :smile:

File: content/en/docs/configuration/acme/dns01/google.md (hunk 1 above)
Comment (meyskens) [reply content/en/docs/configuration/acme/dns01/google.md:28]: I just noticed later on, good point!

File: content/en/docs/configuration/acme/dns01/google.md (hunk 3 above)
Comment (ddgenome) [reply content/en/docs/configuration/acme/dns01/google.md:179]: Yes, I added a blurb about that is my most recent push.

https://github.com/cert-manager/website/pull/306/files#diff-bafc869ae945b0b7647c4ca6a99a0b73R188-R190

**Comment by meyskens** [comment]
@ddgenome fixeed the spelling for you

**Comment by ddgenome** [comment]
Thanks @meyskens , I was just looking into that, trying to figure out where the exclusions were.

**Comment by meyskens** [comment]
/approve
/lgtm
/hold

Thanks this looks great! Holding for merge for today so we can merge it in tomorrow when cert-manager v1.0 is released

**Comment by ddgenome** [comment]
At least I passed alex

**Comment by meyskens** [comment]
We disabled checks on alex... it has way too many false positives :cry:

**Comment by ddgenome** [comment]
Let me rephrase, at least I introduced no new positives, false or otherwise, to alex.

**Comment by ddgenome** [comment]
@meyskens I had to rebase because of conflicts so this PR needs the LGTM label reapplied.
